### PR TITLE
Fix compilation on musl libc

### PR DESCRIPTION
**Pull request by JuniorJPDJ**
I briefly tested it on Alpine Linux and it seems to work.

### CODE REVIEW

File: tcmur_work.c [hunk 1]
```diff
@@ -41,16 +41,11 @@ struct tcmur_work *tcmur_create_work(void)
...
-
-	if (pthread_getname_np(pthread_self(), pname, TCMU_THREAD_NAME_LEN))
-		return;
-
 	/*
 	 * The event work thread may need to do a handler reopen
 	 * call and try to flush itself. Just ignore.
 	 */
-	if (!strcmp(pname, "ework-thread"))
+	if (__tcmu_is_ework_thread)
```
Comment (lxbsz) [inline tcmur_work.c:48]: This seems won't work.
This function will also be call in none `ework-thread` thread.

**Comment by lxbsz** [comment]
And could you split the commit into small ones with each one fixes a single issue. 
At the same time please add what warning or error you are fixing in each commit.
Thanks.

File: tcmur_work.c (hunk 1 above)
Comment (JuniorJPDJ) [reply tcmur_work.c:48]: Could you please elaborate why? I'm not C-coder.

**Comment by JuniorJPDJ** [comment]
> And could you split the commit into small ones with each one fixes a single issue.
At the same time please add what warning or error you are fixing in each commit.
Thanks.

Ok ;) Gimme few minutes.

File: tcmur_work.c (hunk 1 above)
Comment (JuniorJPDJ) [reply tcmur_work.c:48]: Oh. I should negate this. My bad, sorry.

**Comment by JuniorJPDJ** [comment]
[remove assert_perror (musl incompatible)](https://github.com/open-iscsi/tcmu-runner/pull/664/commits/1382a019be7673a5333c8934d18da306ef102540) is for:
```
/home/juniorjpdj/aports/testing/tcmu-runner/tcmu-runner/api.c:77:3: error: implicit declaration of function 'assert_perror'; did you mean 'g_assert_error'? [-Werror=implicit-function-declaration]
   77 |   assert_perror(EINVAL);
      |   ^~~~~~~~~~~~~
```

[add missing pthread.h include](https://github.com/open-iscsi/tcmu-runner/pull/664/commits/d46141f04436fc2f047597b0bcefc4cd40c596da) is for:
```
In file included from /home/juniorjpdj/aports/testing/tcmu-runner/tcmu-runner/configfs.c:19:
/home/juniorjpdj/aports/testing/tcmu-runner/tcmu-runner/libtcmu_common.h:194:25: error: unknown type name 'pthread_t'
  194 | void tcmu_thread_cancel(pthread_t thread);
      |                         ^~~~~~~~~
```

[remove pthread_getname_np calls - musl incompatible](https://github.com/open-iscsi/tcmu-runner/pull/664/commits/91bdeccb201314c24f3ecf92f2113923fb93e435) is for:
```
/home/juniorjpdj/aports/testing/tcmu-runner/tcmu-runner/libtcmu.c:849:6: error: implicit declaration of function 'pthread_getname_np'; did you mean 'pthread_setname_np'? [-Werror=implicit-function-declaration]
  849 |  if (pthread_getname_np(pthread_self(), cname, TCMU_THREAD_NAME_LEN))
      |      ^~~~~~~~~~~~~~~~~~
```
and similar - there was more than one error like this, but point is musl doesn't support this (`pthread_getname_np`) function


[implicitly cast thread id to %lu](https://github.com/open-iscsi/tcmu-runner/pull/664/commits/0a80d8e707d9d9cea141394ad5af822fa6c649fe) is for:
```
/home/juniorjpdj/aports/testing/tcmu-runner/tcmu-runner/libtcmu.c:862:21: error: format '%lu' expects argument of type 'long unsigned int', but argument 5 has type 'pthread_t' {aka 'struct __pthread *'} [-Werror=format=]                                                                                                                                                                                                              
  862 |   tcmu_dev_err(dev, "Failed to set name for thread %lu\n",
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  863 |         pthread_self());
      |         ~~~~~~~~~~~~~~
      |         |
      |         pthread_t {aka struct __pthread *}
```

[fix time_t size in printf to be architecture agnostic](https://github.com/open-iscsi/tcmu-runner/pull/664/commits/df911a2f622b9aebeb0b87b85af0be9cb968af55) is for:
```
src/tcmu-runner-1.5.4/main.c:660:21: error: format '%lu' expects argument of type 'long unsigned int', but argument 5 has type 'time_t' {aka 'long long int'} [-Werror=format=]
  660 |   tcmu_dev_dbg(dev, "Current time %lu secs.\n", time->tv_sec);
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~
      |                                                     |
      |                                                     time_t {aka long long int}
```
and similar (4 errors like this) - this is error while compiling on 32-bit architectures (armhf, armv7, x86).

**Comment by JuniorJPDJ** [comment]
With this patches and one from OpenSuse guy tcmu-runner compiles on musl with 64-bit and 32-bit architectures. To this moment I tested it only on x86_64 and it seem to work but I'll check it on armv7 later today.

**Comment by JuniorJPDJ** [comment]
I reworded commits and added descriptions to them

File: main.c [hunk 2]
```diff
@@ -657,7 +657,7 @@ int tcmur_get_time(struct tcmu_device *dev, struct timespec *time)
 
 	ret = clock_gettime(CLOCK_MONOTONIC_COARSE, time);
 	if (!ret) {
-		tcmu_dev_dbg(dev, "Current time %lu secs.\n", time->tv_sec);
```
Comment (lxbsz) [inline main.c:660]: Yeah, this make sense, while we'd better switch to PRIdMAX macro to make it constant with others in tcmu-runner.

File: tcmur_work.c (hunk 1 above)
Comment (lxbsz) [reply tcmur_work.c:48]: Yeah, you're right.

File: main.c (hunk 2 above)
Comment (JuniorJPDJ) [reply main.c:660]: It's first time I hear about it.
I'll try to do this ;D

Comment (JuniorJPDJ) [reply main.c:660]: Fixed!
I edited commit in place with rebase, nothing else changed.

**Comment by JuniorJPDJ** [comment]
Just tested it on armv7 (32-bit) with fbo backend.
It works very good ;)
image

This means #609 and those commits fix 32-bit issues not only for compilation but also for runtime.
..at least this part I tested

**Comment by lxbsz** [comment]
> Just tested it on armv7 (32-bit) with fbo backend.
> It works very good ;)
> image
> 
> This means #609 and those commits fix 32-bit issues not only for compilation but also for runtime.
> ..at least this part I tested

Cool, I will test it later this week. Thanks @JuniorJPDJ

**Review by lxbsz — APPROVED** [review]

**Comment by JuniorJPDJ** [comment]
Now it would be cool to merge #609 and it would make some packagers not patch tcmu-runner at all ;D

**Comment by lxbsz** [comment]
> Now it would be cool to merge #609 and it would make some packagers not patch tcmu-runner at all ;D

At the same time could help test that ? Currently I didn't have the env, for me I need to install one someday later.

**Comment by JuniorJPDJ** [comment]
I already tested it, it's included in Alpine Linux patches and I was testing it with this pull.

**Comment by lxbsz** [comment]
> I already tested it, it's included in Alpine Linux patches and I was testing it with this pull.

Sounds cool. I will work on it recently after my handy fs work.

**Comment by lxbsz** [comment]
> I already tested it, it's included in Alpine Linux patches and I was testing it with this pull.

Have ever hit the issue @mikechristie mentioned in https://github.com/open-iscsi/tcmu-runner/pull/609/commits/9a4a7211eb27c5b87837cca9211b26ed7baaf8f8#r362956473 ?

**Pull request merged**